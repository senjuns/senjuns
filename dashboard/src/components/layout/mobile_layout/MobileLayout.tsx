import React, { useState } from 'react';
import styled from 'styled-components';
import { useScreenSize } from '../../../hooks/useScreenSize';
import { ScreenSize } from '../../../shared/constants';
import NavigationBar from './NavigationBar';
import SideBar from './SideBar';

const MobileLayout: React.FC = ({ children }) => {
  const [isSideBarOpen, setIsSideBarOpen] = useState(false);
  const { isMobile } = useScreenSize();
  const handleMenuClick = () => {
    setIsSideBarOpen(!isSideBarOpen);
  };

  return (
    <Container isMobile={isMobile}>
      {isSideBarOpen && <SideBar />}
      <PageContainer isSideBarOpen={isSideBarOpen}>
        <NavigationBar
          isSideBarOpen={isSideBarOpen}
          onMenuClick={handleMenuClick}
        />
        <MainArea>{children}</MainArea>
      </PageContainer>
    </Container>
  );
};

interface ContainerProps {
  isMobile: boolean;
}

const Container = styled.div<ContainerProps>`
  position: relative;
  width: 100%;
  ${(props) => (props.isMobile ? `max-width: ${ScreenSize.md}px;` : '')}
  overflow: hidden;
`;

const PageContainer = styled.div<{ isSideBarOpen: boolean }>`
  width: 100%;
  ${(props) => (!props.isSideBarOpen ? 'height: 100vh;' : '')}
  overflow: hidden;
  display: flex;
  flex-direction: column;
  background-color: white;

  @keyframes sliding-down {
    from {
      top: 0;
      bottom: 0;
      left: 0;
      border-radius: 0;
    }
    to {
      top: 10%;
      bottom: 10%;
      left: 300px;
      border-radius: 38px;
    }
  }

  @keyframes sliding-up {
    from {
      top: 10%;
      bottom: 10%;
      left: 300px;
      border-radius: 38px;
    }
    to {
      top: 0;
      bottom: 0;
      left: 0;
      border-radius: 0;
    }
  }

  ${(props) => (props.isSideBarOpen ? 'position: absolute;' : '')}
  left: ${(props) => (props.isSideBarOpen ? '300px' : '0')};
  top: ${(props) => (props.isSideBarOpen ? '10%' : '0')};
  right: 0;
  bottom: ${(props) => (props.isSideBarOpen ? '10%' : '0')};
  border-radius: ${(props) => (props.isSideBarOpen ? '38px' : '0')};
  ${(props) =>
    props.isSideBarOpen
      ? 'box-shadow: -10px 4px 40px rgba(0, 0, 0, 0.08);'
      : ''}
  animation-name: ${(props) =>
    props.isSideBarOpen ? 'sliding-down' : 'sliding-up'};
  animation-duration: 0.3s;
`;

const MainArea = styled.div`
  flex: 1;
  min-width: 0;
  overflow: auto;
`;

export default MobileLayout;
