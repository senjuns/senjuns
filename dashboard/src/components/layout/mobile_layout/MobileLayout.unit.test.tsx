import * as ShallowRenderer from 'react-test-renderer/shallow';
import MobileLayout from './MobileLayout';

describe('<MobileLayout />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<MobileLayout />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
