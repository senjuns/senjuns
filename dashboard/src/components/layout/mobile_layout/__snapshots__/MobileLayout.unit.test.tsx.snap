// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileLayout /> renders correctly. 1`] = `
<styled.div
  isMobile={false}
>
  <styled.div
    isSideBarOpen={false}
  >
    <NavigationBar
      isSideBarOpen={false}
      onMenuClick={[Function]}
    />
    <styled.div />
  </styled.div>
</styled.div>
`;
